File: OCH_Tao/tessoku-book/B13.js
const [[N, K], A] = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n").map(x => x.split(" ").map(y => parseInt(y, 10)));
let B = [0];
for (let i = 0; i < N; i++) {
  B.push(B[B.length - 1] + A[i]);
}
let ans = 0;
let R = 1;
for (let i = 1; i < N + 1; i++) {
  while (R <= N && B[R] - B[i - 1] <= K) {
    R++;
  }
  ans += R - i;
}
console.log(ans);

File: OCH_Tao/tessoku-book/B13.py
from itertools import accumulate
N, K = map(int, input().split())
A = list(map(int, input().split()))
B = list(accumulate([0]+A))
R = [0]*100009
ans = 0

for i in range(1, N+1):
  if i == 1:
    R[i] = 1
  else:
    R[i] = R[i-1]

  while R[i] <= N and B[R[i]]-B[i-1] <= K:
    R[i] += 1

  ans += R[i]-i

print(ans)
